### DISCUSSION TITLE
ScaledObject doesn't have correct Idle/Min/Max Replica Counts specification

### DISCUSSION THREAD

**Discussion by trondhindenes** [kedacore/keda#3014] [Q&A / Need Help]
### Report

Keda reports an error if all of idleReplicaCount, minReplicaCount and maxReplicaCount are 0, effectively disabling the scaler. It isi desirable to be able to disable a scaler instead of deleting it, analoguos to scaling a deployment to zero instead of deleting it.

### Expected Behavior

keda should simply scale its target to 0 if all of idleReplicaCount, minReplicaCount and maxReplicaCount are 0

### Actual Behavior

a warning event is logged: "ScaledObject doesn't have correct Idle/Min/Max Replica Counts specification"

### Steps to Reproduce the Problem

1. create a cron scaler with all 3 attributes mentioned above set to zero
2. notice that a warning is logged

### Logs from KEDA operator

```
example
```

### KEDA Version

2.6.1

### Kubernetes Version

1.21

### Platform

Amazon Web Services

### Scaler Details

cron

**Comment by tomkerkhove** [comment]
Can you elaborate what the scenario is where your max replicas would be 0? Do you want to "pause autoscaling"?

Converting to a discussion because this works as intended, unless I'm misunderstanding the scenario.

**Reply by trondhindenes** [reply]
yup, I'd basically like to pause autoscaling. Our scenario is that we provision a keda scaler for a certain app in all envs, but we might not want to enable autoscaling in all envs. Just changing a number to 0 based on env is way easier than having logic to determine whether or not to create/update/delete the scaler object in our deploy system.

**Reply by tomkerkhove** [reply]
What if you can pause it with a simple annotation? https://keda.sh/docs/2.7/concepts/scaling-deployments/#pause-autoscaling

The bottom line is that KEDA actually works as intended in how you describe it above, but the annotation allows you to achieve what you need.

**Reply by sqaisar** [reply]
I get the same error with 
```
idleReplicas: 0
minReplicas: 1
maxReplicas: 5
```


```
apiVersion: keda.sh/v1alpha1
kind: ScaledObject
metadata:
  name: nm
  namespace: ns
spec:
  fallback:
    failureThreshold: 3
    replicas: 1
  idleReplicaCount: 0
  maxReplicaCount: 5
  minReplicaCount: 1
  pollingInterval: 10
  scaleTargetRef:
    apiVersion: apps/v1
    kind: Deployment
    name: cloud-funcs-worker
  triggers:
    - metadata:
        activationListLength: '1'
        address: >-
          redis-host:6379
        listLength: '1'
        listName: jobs_queue
      type: redis
```yaml

But after deleing it and creating again resolved the issue, I'll try to reproduce and give exact scenario when it reports that error.
